src/core/layertree/qgslayertreemodel.cpp

```diff
@@ -227,6 +227,8 @@ QVariant QgsLayerTreeModel::data( const QModelIndex &index, int role ) const
         if ( testFlag( ShowRasterPreviewIcon ) )
         {
           QgsRasterLayer* rlayer = qobject_cast<QgsRasterLayer *>( layer );
+          if ( rlayer->providerType() == "wms" )
```

Comment:
QStringLiteral( "wms" )

Reply:
It's 2.18, but you can try QLatin1String ;)

Reply:
+1 using QLatin1String would facilitate automatic porting.